packages/amplify-provider-awscloudformation/src/utils/auth-mode-compare.ts

```diff
@@ -0,0 +1,38 @@
+import _ from "lodash";
+
+export function isAuthModeUpdated(options): boolean {
+  const { authConfig, previousAuthConfig } = getAuthConfigForCompare(options);
+  return authConfig && previousAuthConfig && !_.isEqual(authConfig, previousAuthConfig);
```

Comment:
```suggestion
  return !_.isEqual(authConfig, previousAuthConfig);
```

Reply:
We receive authConfig only through `update api` workflow. There are other workflows execute this part like `push` and `gql-compile` which do not pass the authConfig object, so having this condition will make sure that this gets executed only for `update api` workflow.